src/test/java/com/powsybl/openloadflow/ac/AcLoadFlowVscTest.java

```diff
@@ -468,4 +469,64 @@ void testVscVoltageControlWithOneSideDisconnected() {
         assertActivePowerEquals(0, network.getVscConverterStation("cs2").getTerminal());
         assertVoltageEquals(vcs2, network.getVscConverterStation("cs2").getTerminal().getBusView().getBus());
     }
+
+    @Test
+    void testAcEmuWithOperationalLimits() {
+        Network network = HvdcNetworkFactory.createHvdcLinkedByTwoLinesAndSwitch(HvdcConverterStation.HvdcType.VSC);
+        // without limit p=195
+        network.getHvdcLine("hvdc23")
+                .newExtension(HvdcOperatorActivePowerRangeAdder.class)
+                .withOprFromCS2toCS1(180)
+                .withOprFromCS1toCS2(170)
+                .add();
+
+        LoadFlow.Runner loadFlowRunner = new LoadFlow.Runner(new OpenLoadFlowProvider(new DenseMatrixFactory()));
+        LoadFlowParameters p = new LoadFlowParameters();
+        p.setHvdcAcEmulation(true);
+        LoadFlowResult result = loadFlowRunner.run(network, p);
```

Comment:
Is there a test with HvdcAngleDroopActivePowerControl but no HvdcOperatorActivePowerRange extension?

Reply:
@rosiereflo  Yes. All other tests with HVDC in AC emulation. For example testHvdcDirectionChangeAcEmulation

Reply:
Do we have to take into account the active power range extension if the hvdc is operated in set point ?

Reply:
Good question @annetill 
   My approach was to consider that an extension has priority.
   This is consistent with the fact that the AC emulation takes priority on both the mode (side of inverter/rectifier) and active power setPoint.
   But there is nature for conflict because data is duplicated and we need to agree whether giving priority on the extension is what we want or not.
   @rosiereflo

Reply:
Also the fix did not cover the case where AC Emlulation is off.

Reply:
But I have not seen consistency check between p and pmax in HvdcLineAdder

Reply:
In OLF, I have noticed that we can have a generator with a `targetP` over `Pmax`, it is not forbidden but maybe it should... for VSC converter station, no check either. Any idea @geofjamg ?

Reply:
For generators it is very common to have a targetP outside Pmin/Pmax on real data.
Usual expected behavior for these is just to leave them where they are outside Pmin/Pmax and don't change them at all, e.g. exclude them from slack distribution.
I would be against enforcing targetP within Pmin/Pmax for generators in iIDM. Same for reactive power.
For HVDC I don't have any strong opinion though... There aren't that much, and AC emulation is ultra specific to few links.

Reply:
For we can say that it is the user's responsibility and not the purpose of this fix, related to hvdc in AC emulation.